src/odemis/driver/xt_client.py

```diff
@@ -66,6 +71,130 @@
+
+    @path.setter
+    def path(self, value: str) -> None:
+        self._path = value
+
+    @property
+    def version(self) -> str:
+        return self._version
+
+    @version.setter
+    def version(self, value: str) -> None:
+        self._version = value
+
+
+def check_latest_package(
+    directory: str, current_version: str, adapter: str, bitness: str, is_zip: bool
+) -> Optional[Package]:
+    """
+    Check if a latest xtadapter package is available.
+
+    :param directory: The directory to check if a latest xtadapter package is available.
+    :param current_version: The currently installed xtadapter version.
+    :param adapter: The adapter type e.g. xtadapter, fastem-xtadapter to be used for filtering.
+    :param bitness: The executable bitness i.e. 32bit or 64bit to be used for filtering.
+    :param is_zip: A flag stating if the package is a zip file, if not a zip file check if it is a folder.
+    :return: A class containing relevant information about the latest xtadapter package, if not found return None.
+
+    """
+    # Dict where key is the version of xtadapter package and value is information about the package
+    version_package = {}
+    # Package is named as f"delmic-{adapter}-{bitness}bit-v{version}"
+    # delmic-xtadapter-32bit-v1.11.2-dev
+    # delmic-xtadapter-32bit-v1.11.2.zip
+    # delmic-fastem-xtadapter-64bit-v1.11.2-dev
+    regex = r"delmic-([a-zA-Z-]+)-(\d+)bit-v([\d.]+)"
+    if is_zip:
+        regex += r".*.zip"
+    if os.path.isdir(directory):
+        for entity in os.listdir(directory):
+            result = re.search(regex, entity)
+            entity_path = os.path.join(directory, entity)
+            if result and (
+                entity.endswith(".zip") if is_zip
+                else os.path.isdir(entity_path) and len([entity for entity in os.listdir(entity_path) if entity.endswith(".exe")]) == 1
+            ):
+                package = Package()
+                package.adapter = result.group(1)
+                package.bitness = result.group(2) + "bit"
+                package.version = result.group(3)
+                package.name = entity
+                package.path = entity_path
+                # If not a zip file, assign the path of the executable
+                if not is_zip:
+                    for entity in os.listdir(package.path):
+                        if entity.endswith(".exe"):
+                            package.path = os.path.join(package.path, entity)
+                            break
+                # Filter using adapter type and bitness
+                if bitness == package.bitness and adapter == package.adapter:
+                    version_package[package.version] = package
+        versions = list(version_package)
+        if len(versions):
+            # Sort the available versions in reverse order i.e. the latest version is the first element
+            versions.sort(key=lambda s: [int(u) for u in s.split(".")], reverse=True)
```

Comment:
Instead of sorting in reverse order, and then taking the first one, you could just say `newest_version = max(versions, key=...)`. It should be more readable.